[test_sharded_tensor.py] | (Runtime) Integration Tests | runtime-integration-tests / runtime_fd_python_5 [wh_n150_civ2]

## CI Failure Report

<!-- AUTO-TRIAGE-SUMMARY-START -->
**Summary:** [CI] TT_FATAL in test_sharded_tensor: to_layout ROW_MAJOR conversion changes physical shape due to alignment constraints
<!-- AUTO-TRIAGE-SUMMARY-END -->

**Workflow:** (Runtime) Integration Tests
**Job:** runtime-integration-tests / runtime_fd_python_5 [wh_n150_civ2]

### Regression timeline

- **Last passing:** commit `1e5278e4bfb2` — 2026-07-21T05:59:36Z — [run](https://github.com/tenstorrent/tt-metal/actions/runs/29805618193)
- **First failing:** commit `8ae1ef26e2fb` — 2026-07-23T06:00:07Z — [job](https://github.com/tenstorrent/tt-metal/actions/runs/29983702075/job/89211916412)
- **Most recent failing:** commit `6a84cd727e22` — 2026-07-24T05:58:03Z — [job](https://github.com/tenstorrent/tt-metal/actions/runs/30070877721/job/89479299431)

### Failing job URLs (last 2 runs)

- https://github.com/tenstorrent/tt-metal/actions/runs/30070877721/job/89479299431
- https://github.com/tenstorrent/tt-metal/actions/runs/29983702075/job/89211916412

### Failing test path(s)

- `tests/tt_eager/python_api_testing/unit_testing/misc/test_sharded_tensor.py::test_tensor_conversion_between_torch_and_tt_tile[ShardOrientation.ROW_MAJOR-tensor_shape10-TensorMemoryLayout.WIDTH_SHARDED-shard_shape10-grid_override10-DirectReadWriteType.READ_WRITE-DataType.BFLOAT16]`

### Error excerpt (terminal failure from logs)

```
TT_FATAL: to_layout: Converting layout to Layout::ROW_MAJOR implicitly changed physical shape from (32, 16256) to (32, 16384) due to alignment constraints. This is currently unsupported. Please pad the tensor explicitly before conversion.  (assert.hpp:104)
```

```
FAILED tests/tt_eager/python_api_testing/unit_testing/misc/test_sharded_tensor.py::test_tensor_conversion_between_torch_and_tt_tile[ShardOrientation.ROW_MAJOR-tensor_shape10-TensorMemoryLayout.WIDTH_SHARDED-shard_shape10-grid_override10-DirectReadWriteType.READ_WRITE-DataType.BFLOAT16] - RuntimeError: TT_FATAL @ /project/tt_metal/impl/tensor/tensor_apis.cpp:596: output_spec.physical_shape() == tensor.tensor_spec().physical_shape()
info:
to_layout: Converting layout to Layout::ROW_MAJOR implicitly changed physical shape from (32, 16256) to (32, 16384) due to alignment constraints. This is currently unsupported. Please pad the tensor explicitly before conversion.
```

Test parameters:
- `tensor_shape = [1, 1, 32, 16256]`
- `shard_scheme = TensorMemoryLayout.WIDTH_SHARDED`
- `shard_shape = (32, 512)`
- `grid_override = {[0-0 - 4-4], [5-4 - 5-4], [5-5 - 6-6], [6-7 - 7-7]}`

### Reproduction steps

```bash
pytest tests/tt_eager/python_api_testing/unit_testing/ -xvvv --splits 7 --group 5
```

Or to run just the failing test directly:

```bash
pytest "tests/tt_eager/python_api_testing/unit_testing/misc/test_sharded_tensor.py::test_tensor_conversion_between_torch_and_tt_tile[ShardOrientation.ROW_MAJOR-tensor_shape10-TensorMemoryLayout.WIDTH_SHARDED-shard_shape10-grid_override10-DirectReadWriteType.READ_WRITE-DataType.BFLOAT16]" -xvvv
```

Run on a `wh_n150` device (8×8 core grid).

### Notes

The failure occurs in `tt::tt_metal::to_layout` when converting a WIDTH_SHARDED tensor with shape `[1, 1, 32, 16256]` to `Layout::ROW_MAJOR`. The physical shape changes from `(32, 16256)` to `(32, 16384)` due to alignment padding, which is now explicitly rejected by the assertion `output_spec.physical_shape() == tensor.tensor_spec().physical_shape()` at `tensor_apis.cpp:596`.

This regression appears to have been introduced between commits `1e5278e4bfb2` (last passing, 2026-07-21) and `8ae1ef26e2fb` (first failing, 2026-07-23).

**Determinism check:**
- **Run 1** (commit `6a84cd727e22`, 2026-07-24): `FAILED tests/tt_eager/python_api_testing/unit_testing/misc/test_sharded_tensor.py::test_tensor_conversion_between_torch_and_tt_tile[...tensor_shape10...WIDTH_SHARDED...] - RuntimeError: TT_FATAL @ /project/tt_metal/impl/tensor/tensor_apis.cpp:596: output_spec.physical_shape() == tensor.tensor_spec().physical_shape()`
- **Run 2** (commit `8ae1ef26e2fb`, 2026-07-23): `FAILED tests/tt_eager/python_api_testing/unit_testing/misc/test_sharded_tensor.py::test_tensor_conversion_between_torch_and_tt_tile[...tensor_shape10...WIDTH_SHARDED...] - RuntimeError: TT_FATAL @ /project/tt_metal/impl/tensor/tensor_apis.cpp:596: output_spec.physical_shape() == tensor.tensor_spec().physical_shape()`

Both runs show the identical test failure, error class, source location, and shape values.

<!-- AUTO-TRIAGE-METADATA-START -->
`Auto-triage-workflow: (Runtime) Integration Tests`
`Auto-triage-job-name: runtime-integration-tests / runtime_fd_python_5 [wh_n150_civ2]`
<!-- AUTO-TRIAGE-METADATA-END -->

<!-- AUTO-TRIAGE-INFO-START -->
## About this issue (automated CI auto-triage)
This issue was opened and assigned by an automated AI system. It posts through a bot token, so notifications may look like they come from a person, but they are automated.

The owner is selected from recent commit history and job-ownership data. It reflects who is *likely able to fix or route* this, not necessarily who caused it, so it can be imperfect.

**If this was assigned to the wrong person, you can:**
- Comment `REASSIGN OWNER: <why this is outside your area / which team or test-area it belongs to>` and the system will automatically re-route it to a better-suited owner.
- Unassign yourself (this stops notifications to you).
- Close the issue if it is not a real or relevant failure.

The assigned owner is reminded about every 3 days until the issue is resolved. Full guide: https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/2545221692/Auto-Triage+Info
<!-- AUTO-TRIAGE-INFO-END -->

<!-- PIPELINE-REPORT-START -->
## Pipeline report
Analysis report: https://github.com/tenstorrent/tt-metal/actions/runs/30012834703
Slack notification: https://tenstorrent.slack.com/archives/C0B1F8Z7GQM/p1784815112110089?thread_ts=1784814442.443949&cid=C0B1F8Z7GQM
<!-- PIPELINE-REPORT-END -->

--- a/tt_metal/impl/tensor/tensor_apis.cpp
+++ b/tt_metal/impl/tensor/tensor_apis.cpp
@@ -577,23 +577,15 @@ HostTensor to_row_major_layout_impl(const HostTensor& tensor) {
     }
 
     TT_FATAL(tensor.layout() == Layout::TILE, "Converting from {} to Row Major is unsupported.", tensor.layout());
-
-    // Do not pass a non-default tile into ROW_MAJOR PageConfig (deprecation #18536).
-    auto output_spec = TensorSpec(
+    // Construct the new tensor spec first to verify that this is a supported Tensor configuration
+    TensorSpec new_tensor_spec(
         tensor.logical_shape(),
-        TensorLayout(
+        TensorLayout::fromPaddedShape(
             tensor.dtype(),
             PageConfig(Layout::ROW_MAJOR),
-            tensor.memory_config(),
-            tensor.tensor_spec().tensor_layout().get_alignment()));
-
-    TT_FATAL(
-        output_spec.physical_shape() == tensor.tensor_spec().physical_shape(),
-        "to_layout: Converting layout to {} implicitly changed physical shape from {} to {} due to alignment "
-        "constraints. This is currently unsupported. Please pad the tensor explicitly before conversion.",
-        Layout::ROW_MAJOR,
-        tensor.tensor_spec().physical_shape(),
-        output_spec.physical_shape());
+            MemoryConfig{},
+            tensor.logical_shape(),
+            tensor.padded_shape()));
 
     auto tile = tensor.tensor_spec().tile();
     auto physical_shape = tensor.tensor_spec().physical_shape();
@@ -606,9 +598,7 @@ HostTensor to_row_major_layout_impl(const HostTensor& tensor) {
         },
         DistributedHostBuffer::ProcessShardExecutionPolicy::PARALLEL);
 
-    assert_host_shards_match_packed_size(transformed_buffer, output_spec, "to_row_major_layout");
-
-    return HostTensor::from_buffer(std::move(transformed_buffer), output_spec, tensor.tensor_topology());
+    return HostTensor::from_buffer(std::move(transformed_buffer), new_tensor_spec, tensor.tensor_topology());
 }
 
 template <typename T>

